carta/structs.py

```diff
@@ -0,0 +1,16 @@
+import posixpath
+
+from .validation import validate, String, Constant
+from .constants import Polarization, PROTO_POLARIZATION
+
+
+class StokesImage:
+
+    @validate(Constant(PROTO_POLARIZATION), String(), String())
```

Comment:
You don't use `PROTO_POLARIZATION` here -- it's not a validator class; it's just a dict. You use `Polarization` -- the user passes in the same enum object that they would use for polarizations elsewhere.

You get the protobuf index instead of the frontend int value by using the `.proto_index` attribute that we added to each of these enum values above.